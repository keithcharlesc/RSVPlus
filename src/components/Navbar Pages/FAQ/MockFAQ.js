//NOT REQUIRED PAGE
import React, { useState } from "react";
import MockNavigationBar from "../../Navbar Pages/NavigationBar/MockNavigationBar";
import Loop from "./Loop";
import "./FAQ.css";

export default function FAQ() {
  const [faqs, setFaqs] = useState([
    {
      question: "What is RSVPlus?",
      answer:
        "RSVPlus is designed to bring convenience and accessibility to event planners." +
        " It displays upcoming events and reminders, but more importantly it provides an optimal timeslot for group events.",
      open: false,
    },
    {
      question: "Who can use RSVPlus?",
      answer:
        "RSVPlus is first and foremost a calendar app - it is created to be helpful for anyone who wants to keep track of their schedule." +
        " That being said, this app is designed by students for students." +
        " Students are required to work with different teams on projects and co-curricular activities and often do not have an effecient way to decide on a day and time that works with every team member's packed schedule." +
        " RSVPlus solves this problem by providing a date and a time that works well with everybody's schedules.",
      open: false,
    },
    {
      question: "What are channels used for?",
      answer:
        "A channel is created for every event that you organize. First, enter in details about your event like when it starts and when it ends, and who you want to invite." +
        " Once that's done, RSVPlus will automatically create a channel through which you will be able to monitor who has responded to your event." +
        " The people whom you have invited will be able to view the channel via their RSVPlus accounts, and can submit their calendar information to get a date and time that works for everyone after everyone has responded.",
      open: false,
    },
    {
      question:
        "In 'Channels', what happens when I agree to sync my calendar data?",
      answer:
        "When a user clicks on the buton to sync calendar data, RSVPlus identifies specific dates and times that the user is not free." +
        " This is then used alongside other users' calendar data to decide the best date and time that an event can be held.",
      open: false,
    },
    {
      question:
        "In 'Calendar', when I input the link for my calendar, will this link be stored in a database that can be accessed by anyone, including RSVPlus' adminstrators?",
      answer:
        "The calendar link that you submit will be stored in a Firebase database." +
        " However, the link to your calendar cannot be accessed by anyone other than the person to whom it belongs to." +
        " This error message will appear if a person tries to use another person's calendar link: " +
        "'Events from one or more calendars could not be shown here because you do not have the permission to view them.'",
      open: false,
    },
    {
      question:
        "Do RSVPlus store any of my Google Calendar events' private information?",
      answer:
        "Nope. Every time you sync your data to a channel, it only looks for your Google Calendar events that fall within the date range of that channel. Only the needed dates and busy time slots of those dates are stored." +
        " No other information such as event name or details are stored on our database. Solely the needed dates and timings! Furthermore, every time you sync to a new channel, your previous data (old dates and timings) will be cleared and gets updated with the new data for the new date range!",
      open: false,
    },
  ]);

  const toggleFAQ = (index) => {
    setFaqs(
      faqs.map((faq, i) => {
        if (i === index) {
          faq.open = !faq.open;
        } else {
          faq.open = false;
        }

        return faq;
      })
    );
  };

  return (
    <>
      <MockNavigationBar />
      <div className="p-3 mb-2 bg-dark text-white">
        <h2 className="page-header text-center mb-4">FAQ</h2>
        <div
          className="d-flex justify-content-center mt-5"
          style={{ minHeight: "100vh" }}
        >
          <div className="faqs">
            {faqs.map((faq, i) => (
              <Loop key={i} faq={faq} index={i} toggleFAQ={toggleFAQ} />
            ))}
          </div>
        </div>
      </div>
    </>
  );
}
