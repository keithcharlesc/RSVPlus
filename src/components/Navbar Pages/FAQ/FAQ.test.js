import React from "react";
import { render, getByTestId } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { toHaveTextContent } from "@testing-library/jest-dom";
import MockFAQ from "./MockFAQ";

it("renders without crashing", () => {
  const { getByText } = render(<MockFAQ />);
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(<MockFAQ />).toJSON();
  expect(tree).toMatchSnapshot();
});
