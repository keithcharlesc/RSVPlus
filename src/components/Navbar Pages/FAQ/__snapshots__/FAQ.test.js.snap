// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="nav-bar-appName text-danger navbar-brand"
        href="/"
      >
        RSVP+
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/channels"
            href="/channels"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Channels
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/calendar"
            href="/calendar"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calendar
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/faq"
            href="/faq"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            FAQ
          </a>
          <a
            className="nav-bar-text px-3 nav-link"
            data-rb-event-key="/contact-us"
            href="/contact-us"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Contact Us
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="nav-bar-name navbar-text"
          >
            Signed in as: 
            Bob
            <img
              className="photo ml-2 rounded-circle"
              src="thttp://email.com/image"
            />
          </span>
        </div>
      </div>
      <button
        className="ml-3 btn btn-outline-primary"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </nav>
    <div>
      
    </div>
  </div>,
  <div
    className="p-3 mb-2 bg-dark text-white"
  >
    <h2
      className="page-header text-center mb-4"
    >
      FAQ
    </h2>
    <div
      className="d-flex justify-content-center mt-5"
      style={
        Object {
          "minHeight": "100vh",
        }
      }
    >
      <div
        className="faqs"
      >
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            What is RSVPlus?
          </div>
          <div
            className="faq-answer"
          >
            RSVPlus is designed to bring convenience and accessibility to event planners. It displays upcoming events and reminders, but more importantly it provides an optimal timeslot for group events.
          </div>
        </div>
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            Who can use RSVPlus?
          </div>
          <div
            className="faq-answer"
          >
            RSVPlus is first and foremost a calendar app - it is created to be helpful for anyone who wants to keep track of their schedule. That being said, this app is designed by students for students. Students are required to work with different teams on projects and co-curricular activities and often do not have an effecient way to decide on a day and time that works with every team member's packed schedule. RSVPlus solves this problem by providing a date and a time that works well with everybody's schedules.
          </div>
        </div>
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            What are channels used for?
          </div>
          <div
            className="faq-answer"
          >
            A channel is created for every event that you organize. First, enter in details about your event like when it starts and when it ends, and who you want to invite. Once that's done, RSVPlus will automatically create a channel through which you will be able to monitor who has responded to your event. The people whom you have invited will be able to view the channel via their RSVPlus accounts, and can submit their calendar information to get a date and time that works for everyone after everyone has responded.
          </div>
        </div>
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            In 'Channels', what happens when I agree to sync my calendar data?
          </div>
          <div
            className="faq-answer"
          >
            When a user clicks on the buton to sync calendar data, RSVPlus identifies specific dates and times that the user is not free. This is then used alongside other users' calendar data to decide the best date and time that an event can be held.
          </div>
        </div>
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            In 'Calendar', when I input the link for my calendar, will this link be stored in a database that can be accessed by anyone, including RSVPlus' adminstrators?
          </div>
          <div
            className="faq-answer"
          >
            The calendar link that you submit will be stored in a Firebase database. However, the link to your calendar cannot be accessed by anyone other than the person to whom it belongs to. This error message will appear if a person tries to use another person's calendar link: 'Events from one or more calendars could not be shown here because you do not have the permission to view them.'
          </div>
        </div>
        <div
          className="faq "
          onClick={[Function]}
        >
          <div
            className="faq-question"
          >
            Do RSVPlus store any of my Google Calendar events' private information?
          </div>
          <div
            className="faq-answer"
          >
            Nope. Every time you sync your data to a channel, it only looks for your Google Calendar events that fall within the date range of that channel. Only the needed dates and busy time slots of those dates are stored. No other information such as event name or details are stored on our database. Solely the needed dates and timings! Furthermore, every time you sync to a new channel, your previous data (old dates and timings) will be cleared and gets updated with the new data for the new date range!
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
